### DISCUSSION TITLE
How to react to external events?

### DISCUSSION THREAD

**Discussion by eduardoarandah** [alpinejs/alpine#1284] [1. Help]
I have a file uploader and a file list

[uploader]

[alpine file list]

I'd like to inform alpine that it should update when dropzone uploader uploads a file

however, if I call `myalpine().fetch();` nothing happens. 

Any ideas? 

example: 

```
     function myalpine() {
        return {
          fetch() {
            fetch(url).then(r => r.json()).then(r => this.files = r);
          },
        }
      }

      Dropzone.autoDiscover = false;
      new Dropzone("#dropzone", {
        init: function() {
          this.on("complete", function() {
            myalpine().fetch();
          });
        }
      });

```

**Comment by ryangjchandler** [comment]
The function `myalpine` has no relation to your component at all. You'd be better of using `x-init`, initialising the `Dropzone` instance inside of the `x-init` function and that way, you have full access to your component.

Let me know if you need some assistance with the syntax, etc.

**Reply by eduardoarandah** [reply]
Thanks. Solved with a global event listener. 

Check it out:

https://gist.github.com/eduardoarandah/b04ebb27c46ccf6e70e9e17c7667f9f1
